src/helper.rs

```diff
@@ -8,64 +9,94 @@ use thiserror::Error;
 
 use crate::schema::Nftables;
 
+/// Default `nft` executable.
 const NFT_EXECUTABLE: &str = "nft"; // search in PATH
 
+/// Use the default `nft` executable.
+pub const DEFAULT_NFT: Option<&str> = None;
+
+/// Do not use additional arguments to the `nft` executable.
+pub const DEFAULT_ARGS: Option<&[&str]> = None;
+
+/// Error during `nft` execution.
 #[derive(Error, Debug)]
 pub enum NftablesError {
-    #[error("unable to execute {program}: {inner}")]
-    NftExecution { program: String, inner: io::Error },
-    #[error("{program}'s output contained invalid utf8: {inner}")]
+    #[error("unable to execute {program:?}: {inner}")]
+    NftExecution { program: OsString, inner: io::Error },
+    #[error("{program:?}'s output contained invalid utf8: {inner}")]
     NftOutputEncoding {
-        program: String,
+        program: OsString,
         inner: FromUtf8Error,
     },
     #[error("got invalid json: {0}")]
     NftInvalidJson(serde_json::Error),
-    #[error("{program} did not return successfully while {hint}")]
+    #[error("{program:?} did not return successfully while {hint}")]
     NftFailed {
-        program: String,
+        program: OsString,
         hint: String,
         stdout: String,
         stderr: String,
     },
 }
 
-pub fn get_current_ruleset(
-    program: Option<&str>,
-    args: Option<&[&str]>,
+/// Get the rule set that is currently active in the kernel.
+///
+/// This is done by calling the default `nft` executable with default arguments.
+pub fn get_current_ruleset() -> Result<Nftables<'static>, NftablesError> {
+    get_current_ruleset_with_args(DEFAULT_NFT, DEFAULT_ARGS)
+}
+
+/// Get the current rule set by calling a custom `nft` with custom arguments.
+///
+/// If `program` is [Some], then this program will be called instead of the
+/// default `nft` executable.
+/// [DEFAULT_NFT] can be passed to call the default `nft`.
+///
+/// If `args` is [Some], then these `nft` arguments will be used instead of the
+/// default arguments `list` and `ruleset`.
+/// [DEFAULT_ARGS] can be passed to use the default arguments.
+/// Note that the argument `-j` is always added in front of `args`.
+pub fn get_current_ruleset_with_args<P: AsRef<OsStr>, A: AsRef<OsStr>>(
+    program: Option<P>,
+    args: Option<&[A]>,
 ) -> Result<Nftables<'static>, NftablesError> {
```

Comment:
Could this (and friends) be more generalized like the following, such that `Option<&A>` and containers of `&A` can both be passed?

```rust
pub fn get_current_ruleset_with_args<'a, P, A, I>(
    program: Option<&P>,
    args: I,
) -> Result<Nftables<'static>, NftablesError>
where
    P: AsRef<OsStr> + ?Sized,
    A: AsRef<OsStr> + ?Sized + 'a,
    I: IntoIterator<Item = &'a A> + 'a,
```

I've integrated this in my own fork: https://github.com/nftables-rs/nftables-rs/commit/3e3230a187213455d6df1072ee3b4f5e39f80ef3.

(not maintainer, just have some API design ideas)

Reply:
Yeah. Especially not using Option here and instead using an empty iterator (or slice, for that matter) would make some sense, IMO.
About the iterator: Yes, I guess that makes some sense. Especially because Command does it like this, too.

Reply:
Thanks, Eric. I have included your suggestion in the PR and I have also rebased on latest main branch. I have just changed the type of the DEFAULT_ARGS constant.